src/app/Shared/Services/Report.service.tsx

```diff
@@ -43,7 +43,12 @@ export class ReportService {
       headers,
     }).pipe(
       concatMap((resp) => {
+        // 200 indicates that the backend has the report cached and it will return
+        // the json in the response.
         if (resp.ok) {
+          if (resp.status == 202) {
+            this.notifications.info('Report generation in progress', 'Report is being generated');
+          }
```

Comment:
I think in the case that we get a 200 response, we want to check the job UUID sent back by the server and hang on to that somehow. Maybe something as simple as a list of "report job UUIDs" that we append to.